### ISSUE TITLE
Enhancement: Add OpenCycle Overlay

### ISSUE THREAD

**Issue by magdesign** [janbar/osmin#22]
Would be very nice to have  [OpenCycle Map](https://www.opencyclemap.org/) overlay under **Style**

https://wiki.openstreetmap.org/wiki/Cycle_routes

**Comment by janbar** [comment]
In the menu "Configure Map" you can select others styles. The style "outdoor" looks like cycle map. isn't it ?

**Comment by janbar** [comment]
Need to add a new stylesheet (`resources/stylesheets/`), starting by a copy of outdoor or standard. I have to check if @karry made already a stylesheet for cycle.

**Comment by magdesign** [comment]
To try on my end, am I right that I would have to change [outdoor.oss](https://github.com/janbar/osmin/blob/469eb98cbcf3ddb002fb3f2ab0b74636075c579f/resources/stylesheets/outdoor.oss)

and edit from here:   https://github.com/janbar/osmin/blob/469eb98cbcf3ddb002fb3f2ab0b74636075c579f/resources/stylesheets/outdoor.oss#L2074


[cycle_network]()=*
might be the solution

**Comment by Karry** [comment]
Hi guys. It seems to me, that what you want is to distinguish routes with [icn](https://wiki.openstreetmap.org/wiki/Key:icn)=yes (international), [ncn](https://wiki.openstreetmap.org/wiki/Key:ncn)=yes (national), [rcn](https://wiki.openstreetmap.org/wiki/Key:rcn)=yes (regional) and [lcn](https://wiki.openstreetmap.org/wiki/Key:lcn)=yes (local) tags... Not sure how these tags are adopted in different countries. But OSM Scout maps doesn't import this information now. We just distinguish road bike routes and mountain bike routes:
```
  TYPE route_bicycle
    = RELATION ("type"=="route" AND "route"=="bicycle")
      {Name, Ref, Operator, Network, Color}
      ROUTE

  TYPE route_mtb
    = RELATION ("type"=="route" AND "route"=="mtb")
      {Name, Ref, Operator, Network, Color}
      ROUTE
``` 
So first, you would need to extend import definition here: https://github.com/Framstag/libosmscout/blob/master/stylesheets/map.ost#L2376 and then you can start improving stylesheet...

**Comment by janbar** [comment]
Thanks Karry, by trying to show on the map the needed relation: "network"=="lcn"
![image](https://user-images.githubusercontent.com/2353635/163226884-f63e2271-0b32-4a17-92c8-f47197411166.png)

**Comment by janbar** [comment]
But updating the ost file to split "route_bicycle" in 3 new types i.e "route_bicycle_lcn", "route_bicycle_ncn" and "route_bicycle_icn" , we have to build new maps.
Instead we could create a new stylesheet "cycle.oss" to highlight route_bicycle (but only one color regardless the network value).

**Comment by magdesign** [comment]
create a new stylesheet "cycle.oss" sounds like a good solution.
I think the color is not very important, but to fetch the number of the cycle route would be great (if possible).

**Comment by Framstag** [comment]
I just want to note, that using multiple style sheets on the same database was one of the main goals of libosmscout.  Yes, this requires that the database holds all types required by all style sheets, so adding types are splitting types might be necessary - breaking compatibility.
Note also that while style sheets do not have includes (one could add such feature) you can generate variants (like we do for day/night versions) and one could generate stylesheets from modular "snippets" . 
Using such technics, one could generate multiple stylesheets with individual specializations or layers from base styles.
Storing the stylesheets in the upstream libosmscout projects has the advantage that they can reuse between applications and that we can test such stylesheets,
IF you are missing visualization primitives, please make a ticket :-)

**Comment by janbar** [comment]
Yes it is possible to fetch the number. In my previous preview I printed the network.name. Instead we can print the ref.name which is the number.
I think to start from the winter-sports.oss to create the cycle.oss.

**Comment by Karry** [comment]
What is actually missing in the outdoor map? Outdoor style shows cycle route "ref" already ... `ROUTE.TEXT {label: Ref.name` I am not against new specialized cycle map, just curious about your requirements. I love road cycling and outdoor style  (originally created by Vlad, I just added routes on top of it) is enough for me usually.

> But updating the ost file to split "route_bicycle" in 3 new types i.e "route_bicycle_lcn", "route_bicycle_ncn" and "route_bicycle_icn" , we have to build new maps.

It is not so big issue IMO, I rebuild whole world monthly usually. Question is if this "cycle network hierarchy" brings some additional information to the user. In Czech, route quality and infrastructure around it is usually the same...

And speaking about updating stylesheet, you know [StyleEditor](https://github.com/Framstag/libosmscout/tree/master/StyleEditor), right? :-)

**Comment by janbar** [comment]
I post a stylesheet "cycle.oss" for **road** cycle (no mountain bike).

[cycle.zip](https://github.com/janbar/osmin/files/8490355/cycle.zip)

@Karry , the outddor map is right for me. I guess @magdesign would like a stylesheet for bike path only. The joined oss file should resolve the need.

**Comment by Karry** [comment]
Looks nice :-) Just route labels are poorly visible. And it would be great to see international routes even on low zoom...

**Comment by janbar** [comment]
@Karry , I updated the style, but attributes are limited for a path text.
[cycle.zip](https://github.com/janbar/osmin/files/8495082/cycle.zip)

**Comment by Framstag** [comment]
I have now added includes to OSR and OSS files. I added some sample modules, but do not have plans myself to further modularize current styles. I suggest you to team up with @Karry for experiments in this area. Together with the already implemented conditionals, this should help to add more variants of style easily. It may also help with #10. Note that I had to increase the NDK version (for the also new use for std::filesytem).

Have fun :-)
